cli/central-client/src/main/java/org/ballerinalang/central/client/CentralAPIClient.java

```diff
@@ -896,7 +905,16 @@ protected void handleResponseErrors(Response response, String msg) throws Centra
                     response.code() == HTTP_UNAVAILABLE) {
                 Error error = new Gson().fromJson(body.get().string(), Error.class);
                 if (error.getMessage() != null && !"".equals(error.getMessage())) {
-                    throw new CentralClientException(msg + "' reason:" + error.getMessage());
+                    throw new CentralClientException(msg + " reason:" + error.getMessage());
+                }
+            }
+
+            // If invalid/no response received at the gateway server
+            if (response.code() == HTTP_BAD_GATEWAY ||
```

Comment:
What happens if an HTTP response code other than the handled ones are received?

Reply:
It throws an CentralClientException with the provided error message. 

Line 922. https://github.com/gayaldassanayake/ballerina-lang/blob/eb0f3abb4f6268f95e1ad420eee5f9b99e0cf752/cli/central-client/src/main/java/org/ballerinalang/central/client/CentralAPIClient.java#L922.

Will add a test case representing this scenario.